Framework/DataHandling/test/SaveGDATest.h

```diff
@@ -0,0 +1,308 @@
+      numPoints += static_cast<int>(lineItems.size()) / 3;
+      numLines++;
+
+      for (size_t i = 0; i < lineItems.size(); i += 3) {
+        TOFs.emplace_back(std::stod(lineItems[i]));
+      }
+    }
+
+    const auto expectedNumPoints = std::stoi(headerItems[2]);
+    TS_ASSERT_EQUALS(expectedNumPoints, numPoints);
+
+    const auto expectedNumLines = std::stoi(headerItems[3]);
+    TS_ASSERT_EQUALS(expectedNumLines, numLines);
+
+    const auto expectedTOFMin1 = std::stoi(headerItems[5]);
+    TS_ASSERT_EQUALS(TOFs[0], expectedTOFMin1);
+
+    const auto expectedTOFMin2 = std::stoi(headerItems[7]);
+    TS_ASSERT_EQUALS(TOFs[0], expectedTOFMin2);
+
+    const auto averageDeltaTByT = computeAverageDeltaTByT(TOFs);
+    const auto expectedAverageDeltaTByT = std::stod(headerItems[8]);
+    TS_ASSERT_DELTA(expectedAverageDeltaTByT, averageDeltaTByT, 1e-3);
+
+    // Just make sure there's another header after the one we just checked
+    TS_ASSERT(boost::starts_with(line, "BANK 1"));
+  }
+
+  void test_dataIsCorrect() {
+    SaveGDA testAlg;
+    testAlg.initialize();
+    testAlg.setProperty("InputWorkspace", INPUT_GROUP_NAME);
+    testAlg.setProperty("GSASParamFile", m_paramsFile.path());
+    testAlg.setProperty("GroupingScheme", std::vector<int>({2, 3}));
+    Poco::TemporaryFile tempFile;
+    const std::string &tempFileName = tempFile.path();
+    testAlg.setProperty("Filename", tempFileName);
+    testAlg.execute();
+
+    std::ifstream file(tempFileName);
+    std::string line;
+    TS_ASSERT(file.is_open());
+
+    // first line is header
+    std::getline(file, line);
+
+    std::vector<int> tof;
+    std::vector<int> intensity;
+    std::vector<int> error;
+    while (std::getline(file, line) && !boost::starts_with(line, "BANK")) {
+      std::vector<std::string> lineItems;
+      boost::trim(line);
+      boost::split(lineItems, line, boost::is_any_of(" "),
+                   boost::token_compress_on);
+      for (size_t i = 0; i < lineItems.size(); i += 3) {
+        TS_ASSERT_THROWS_NOTHING(tof.emplace_back(std::stoi(lineItems[i])));
+        TS_ASSERT_THROWS_NOTHING(
+            intensity.emplace_back(std::stoi(lineItems[i + 1])));
+        TS_ASSERT_THROWS_NOTHING(
+            error.emplace_back(std::stoi(lineItems[i + 2])));
+      }
+    }
+
+    const static size_t expectedNumPoints = 13000;
+    TS_ASSERT_EQUALS(tof.size(), expectedNumPoints);
+    TS_ASSERT_EQUALS(intensity.size(), expectedNumPoints);
+    TS_ASSERT_EQUALS(error.size(), expectedNumPoints);
+
+    // Test a few reference values
+    TS_ASSERT_EQUALS(tof[103], 99772);
+    TS_ASSERT_EQUALS(intensity[103], 1);
+    TS_ASSERT_EQUALS(error[103], 34);
+
+    TS_ASSERT_EQUALS(tof[123], 100725);
+    TS_ASSERT_EQUALS(intensity[123], 1);
+    TS_ASSERT_EQUALS(error[123], 35);
+
+    TS_ASSERT_EQUALS(tof[3000], 239053);
+    TS_ASSERT_EQUALS(intensity[3000], 800);
+    TS_ASSERT_EQUALS(error[3000], 894);
```

Comment:
Same for all of these arrays, what happens if there are not 3000 elements (due to a regression in the original algorithm)

Reply:
This is already taken care of in lines 225 to 228